ui/app/routes/observability/inferences/$inference_id/VariantResponseModal.tsx

```diff
@@ -63,12 +63,16 @@ export function VariantResponseModal({
       setVariantResponse(null);
       setShowRawResponse(false);
       setError(null);
-      const request = {
-        function_name: inference.function_name,
-        input: inference.input,
-        variant_name: selectedVariant,
-        dryrun: true,
-      };
+      const request =
+        inference.function_name === "tensorzero::default"
+          ? prepareDefaultFunctionRequest(inference, selectedVariant)
+          : {
+              function_name: inference.function_name,
+              input: inference.input,
+              variant_name: selectedVariant,
+              dryrun: true,
+            };
+
       variantInferenceFetcher.submit(
         { data: JSON.stringify(request) },
```

Comment:
why are you `JSON.stringify` the request? this seems unnecessary and something that could be unwound

Reply:
I had trouble getting the form data the fetcher submits to play well with nested objects --- so I figured I'd stringify and then parse / validate on the other end. If I'm missing the correct way to do it I'd love to learn.